src/Core/Compiler/NodeConverter.php

```diff
@@ -367,9 +375,36 @@ protected function convertExpressionNode(ExpressionNodeInterface $node)
      */
     protected function convertBooleanNode(BooleanNode $node)
     {
-        $stack = $this->convertArrayNode(new ArrayNode($node->getStack()));
-        $initializationPhpCode = '// Rendering Boolean node' . chr(10);
-        $initializationPhpCode .= $stack['initialization'] . chr(10);
+        $booleanStack = $node->getStack();
+
+        // Quick decisions: if there is only one node and it can be determined to be a string/numeric/boolean already
+        // then use it directly.
+        if (count($booleanStack) === 1) {
```

Comment:
this area has been marked as an @todo with 9785955cf59014dcba234f921d5d150bfdc1eec9